### PR TITLE
Update docs + error message when adding local users

### LINKED ISSUE #3083: Provide helpful error message when adding local users (teleport - enterprise)
**What happened:** User was trying to create a user for a k8s group with the following command

`sudo tctl users add joe --k8s-groups="system:masters"`

Error recieved was:

`error: unknown long flag '--k8s-groups'`

**What you expect to happen:**

Whenever we add a local user in enterprise, and a user tries to use `--k8s-groups` or `--logins` error handling should provide a helpful error message instructing the user to use RBAC and create a role with k8s groups. When creating a local user, simply use `--roles` flag. 

Also update documentation [here](https://gravitational.com/teleport/docs/admin-guide/#kubernetesteleport-configuration) and add note mentioning to create a k8s group.

**References:**

Add creates a new sign-up token and prints token URL to stdout [here](https://github.com/gravitational/teleport/blob/master/tool/tctl/common/user_command.go#L105)

Function `TryRun` takes CLI command as an argument and executes it [here](https://github.com/gravitational/teleport/blob/master/tool/tctl/common/user_command.go#L87)

Function `ParseRoles` takes list of roles and returns a slice of roles, or an error if parsing failed [here](https://github.com/gravitational/teleport/blob/master/roles.go#L75)

**Related issue:** #2950

### PR DESCRIPTION
**Pull request by aelkugia**
This PR updates docs when adding local users + error handling for when user adds role flag that does not exist. Several customers have ran into this issue, and current error reporting doesn't clarify cause of issue.

Resolves: #3083

### CODE REVIEW

File: docs/4.1/admin-guide.md [hunk 1]
```diff
@@ -1991,6 +1991,9 @@ belong to:
 $ tctl users add joe --k8s-groups="system:masters"
 ```
 
+!!! tip "NOTE":
+    In order to execute the above command, a Teleport role in RBAC needs to be created as `k8s-group`. An example RBAC role can be found in the [Kubernetes SSH](https://gravitational.com/teleport/docs/kubernetes_ssh/#impersonation) section.
```
Comment (benarent) [inline docs/4.1/admin-guide.md]: For internal links we are using mkdocs functionally, so we can run automated tests on links.

This should be 
`kubernetes_ssh.md#impersonation`

**Review by benarent — CHANGES_REQUESTED** [review]

File: docs/4.1/admin-guide.md (hunk 1 above)
Comment (aelkugia) [reply docs/4.1/admin-guide.md]: That's right, just committed this update

**Comment by webvictim** [comment]
ok to test

**Review by benarent — APPROVED** [review]

File: tool/tctl/common/user_command.go [hunk 2]
```diff
@@ -118,6 +118,9 @@ func (u *UserCommand) Add(client auth.ClientI) error {
 	}
 	token, err := client.CreateSignupToken(user, u.ttl)
 	if err != nil {
+		if strings.Contains(err, "unknown long flag") {
+			return trace.Wrap(err, "NOTE: Make sure role used in flag has a corresponding Teleport role in RBAC")
```
Comment (russjones) [inline tool/tctl/common/user_command.go]: @aelkugia What does this output when `tctl` fails?

File: docs/4.1/admin-guide.md [hunk 3]
```diff
@@ -1991,6 +1991,9 @@ belong to:
 $ tctl users add joe --k8s-groups="system:masters"
 ```
 
+!!! tip "NOTE":
+    In order to execute the above command, a Teleport role in RBAC needs to be created as `k8s-group`. An example RBAC role can be found in the [Kubernetes SSH](kubernetes_ssh.md#impersonation) section.
```
Comment (webvictim) [inline docs/4.1/admin-guide.md]: Is this saying you need a role specifically called `k8s-group` in Kubernetes RBAC? Referring to some kind of `ClusterRole` or `ClusterRoleBinding` that will enable impersonation inside Kubernetes? It's not incredibly clear currently.

Comment (aelkugia) [reply docs/4.1/admin-guide.md]: Reword to something along the following:

The above command is for the OSS version, Enterprise customers manage k8s-groups through RBAC

Comment (benarent) [reply docs/4.1/admin-guide.md]: @aelkugia Can you add this to 4.2 docs also.

**Review by benarent — CHANGES_REQUESTED** [review]

File: docs/4.1/admin-guide.md (hunk 3 above)
Comment (aelkugia) [reply docs/4.1/admin-guide.md]: I'm just going to close this PR as it is pretty out of date, and in discussion with @fspmarshall the below update of where I was handling the error was incorrect as well. I can add updates to docs to 4.1 and 4.2 in the new PR which I will create

**Comment by webvictim** [comment]
Thanks @aelkugia - sorry it took so long to review this. In future we’ll pay better attention.

**Pull request closed without merging**